documentation/docs/tutorials/single-topic-multi-tenant.mdx

```diff
@@ -0,0 +1,341 @@
+---
+sidebar_position: 12
+title: Single-Topic Multi-Tenant Configuration
+---
+
+# Single-Topic Multi-Tenant Configuration
+
+This document describes how to configure OPAL for multi-tenant environments using a single topic approach. This configuration allows adding new tenants without restarting OPAL services.
+
+## Understanding Single-Topic Multi-Tenancy
+
+### Traditional multi-tenant configuration
+
+In traditional OPAL setups, each tenant requires its own data topic:
+
+```
+Tenant 1: OPAL_DATA_TOPICS=tenant_1_data
+Tenant 2: OPAL_DATA_TOPICS=tenant_1_data,tenant_2_data  # requires restart
+Tenant 3: OPAL_DATA_TOPICS=tenant_1_data,tenant_2_data,tenant_3_data  # requires restart
+```
+
+This traditional approach creates several challenges:
+
+- **Event-driven vs polling limitations** - When you need event-driven data updates instead of interval-based polling, managing multiple topics becomes complex and resource-intensive.
+
+- **Bundle scalability issues** - For thousands of tenants with large datasets, updating data through undivided bundles becomes impractical. This forces you to implement data sharding or tenant-specific bundle divisions, which are difficult to prepare and maintain.
+
+- **State management complexity** - Divided bundles introduce problems with state management, particularly around bundle ordering and ensuring consistent data application across tenants.
+
+- **Topic proliferation** - Achieving tenant isolation through separate data sources results in an ever-growing number of topics, making the system difficult to manage and scale.
+
+- **Configuration management overhead** - Each new tenant requires configuration changes and service restarts, making dynamic tenant onboarding cumbersome.
+
+The single-topic configuration described below addresses these issues by providing a simpler, more scalable approach.
+
```

Comment:
Worth adding a diagram to explain the architecture

```suggestion

```mermid
flowchart
    subgraph "Multi-scope"
        DS1["Data source: Tenant 1"]
        DS2["Data source: Tenant 2"]
        DSN["Data source: Tenant N"]
        DS1 --> OC1["OPAL Client: Scope 1"]
        DS2 --> OC2["OPAL Client: Scope 2"]
        DSN --> OCN["OPAL Client: Scope N"]
        OC1 <--> SVC["Application services"]
        OC2 <--> SVC
        OCN <--> SVC
    end
```

```